internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -596,25 +628,38 @@ func resourceKeyVaultCertificateUpdate(d *schema.ResourceData, meta interface{})
 	return resourceKeyVaultCertificateRead(d, meta)
 }
 
-func keyVaultCertificateCreationRefreshFunc(ctx context.Context, client *keyvault.BaseClient, keyVaultBaseUrl string, name string) pluginsdk.StateRefreshFunc {
+func keyVaultCertificateCreationRefreshFunc(ctx context.Context, client *keyvault.BaseClient, keyVaultBaseUrl string, name string, requestID string) pluginsdk.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		res, err := client.GetCertificate(ctx, keyVaultBaseUrl, name, "")
+		operation, err := client.GetCertificateOperation(ctx, keyVaultBaseUrl, name)
 		if err != nil {
-			return nil, "", fmt.Errorf("issuing read request in keyVaultCertificateCreationRefreshFunc for Certificate %q in Vault %q: %s", name, keyVaultBaseUrl, err)
+			return nil, "", fmt.Errorf("failed to read CertificateOperation in keyVaultCertificateCreationRefreshFunc for Certificate %q in Vault %q: %s", name, keyVaultBaseUrl, err)
+		}
+		if operation.RequestID == nil {
+			return nil, "", fmt.Errorf("missing request_id in certificate operation")
+		}
+
+		// If another process starts a new certificate creation right after our
+		// operation was completed we might not observe the completed state. Also as
+		// soon someone started a new create operation there is no way to determine
+		// what happened to the previous operation. Because of that we return an error
+		// if there is a new operation (not the same request ID) running.
+		if *operation.RequestID != requestID {
+			return nil, "", fmt.Errorf("could not observe outcome of certificate creation because another process started a new creation")
+		}
+
+		if operation.Status == nil {
+			return nil, "", fmt.Errorf("missing status in certificate operation")
 		}
 
-		if res.Policy != nil &&
-			res.Policy.IssuerParameters != nil &&
-			res.Policy.IssuerParameters.Name != nil &&
-			strings.EqualFold(*(res.Policy.IssuerParameters.Name), "unknown") {
-			return res, "Ready", nil
+		if *operation.Status == "inProgress" {
```

Comment:
This should be checked without casing in case there are discrepancies between the api and what we expect 

```suggestion
		if strings.EqualFolds(*operation.Status, "inProgress") {
```

Reply:
@mbfrahry applied your suggestion. don't really understand what you mean by discrepancies? usually I think it is better if we are explict about what we expect? but probably doesn't really matter